src/librustc_codegen_ssa/back/write.rs

```diff
@@ -1468,6 +1479,18 @@ fn start_executing_work<B: ExtraBackendMethods>(
                     // Note, however, that this is not ideal for memory
                     // consumption, as LLVM module sizes are not evenly
                     // distributed.
+
+                    // njn: does the above sorting penalize Metadata modules?
+
+                    // njn: This is my best guess about when to emit the
+                    // directive, but I could be wrong. Is the .rmeta file
+                    // guaranteed to be written at this point? I don't know...
+                    // I couldn't find where the file writing occurs.
+                    if let WorkItem::Optimize(
+                        ModuleCodegen { name, kind: ModuleKind::Metadata, .. }) = &llvm_work_item {
+                        cgcx.diag_emitter.emit_directive(format!("metadata done for {}", name));
```

Comment:
I think we should emit a message for *all* artifacts, with a path to the artifact, and let Cargo pick out what it cares out of that.
Something like sccache, for example, might care more about other files.
And it's less tied to pipelining, just another general thing rustc does.